src/AutoMapper/QueryableExtensions/ProjectionBuilder.cs

```diff
@@ -165,11 +165,10 @@ Expression MapFromExpression(LambdaExpression mapFrom)
                     {
                         if (memberTypeMap == null || mapFrom.IsMemberPath(out _) || mapFrom.Body is ParameterExpression)
                         {
-                            var parmaterExtractor = new ParameterExpressionExtractorVisitor();
-                            parmaterExtractor.Visit(mapFrom.Body);
-                            if (parmaterExtractor.Parameter != null)
+                            var member = MemberAccessExpressionExtractorVisitor.Extract(mapFrom.Body, mapFrom.Parameters.First());
+                            if (member != null)
```

Comment:
What does this actually mean? Why would accessing smth on the parameter matter?

Reply:
the code needs to decide weather a type conversion is required.

before I used the parameter type to detect this.

but that resulted in unnessesay conversions in case the parameter is of the derived type but the used property is defined in the base class.  

the new code only adds a conversion if the member of the parameter is actually declared on the derived type

Reply:
Ahh I was not thinking about using multiple members in one expression tree I will update the code

Reply:
done. i now extract all member expressions on the given parameter and use the most derived type to determine weather a conversion is required.